### DISCUSSION TITLE
IPipelineBehavior doesn't work with IRequest

### DISCUSSION THREAD

**Discussion by GeorgeKarlinzer** [LuckyPennySoftware/MediatR#968] [Q&A]
`IPipelineBehavior` cannot be used with `IRequest` (not `IRequest<out TResponse>`), because `IPipelineBehavior` requires `TResponce` as well as `TRequest`, but `IRequest` does not have a return value. Is it possible to add a pipeline behavior for `IRequest`?

https://github.com/jbogard/MediatR/blob/c295291c4e8105d11a453004b42609cbf490c1cf/src/MediatR/IPipelineBehavior.cs#L20-L30

https://github.com/jbogard/MediatR/blob/c295291c4e8105d11a453004b42609cbf490c1cf/src/MediatR.Contracts/IRequest.cs#L6-L17

**Comment by jbogard** [comment]
Pipeline behaviors DO work with void requests, and my unit tests show it. I suspect you may not have adjusted your generic constraints on one of the recent releases, you may want to re-check that.

**Reply by GeorgeKarlinzer** [reply]
You are right, I triple checked my code and it was my problem, thank you
